src/routers/StackRouter.js

```diff
@@ -139,7 +139,7 @@ export default (
       }
 
       // Check if a child scene wants to handle the action as long as it is not a reset to the root stack
-      if (action.type !== NavigationActions.RESET || action.key !== null) {
+      if (action.type !== NavigationActions.RESET || !!action.key) {
```

Comment:
Agreed with this -- let's explicitly check for `undefined` in addition to `null`.